### PR TITLE
Run3-alca192 Add a few methods in Calibration/IsolatedParticles

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a few methods in Calibration/IsolatedParticles

#### PR validation:

Tested using cfg's specific for AlCa cfg's in Calibrartion/HcalCalibAlgos/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
Hi @bsunanda can you please remove the cout-s? Thanks!

**Comment by bsunanda** [comment]
@tvami I shall remove cout statements later in a new PR once this PR is integrated. I shall only take care of clang warnings now

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@tvami Please approve this

**Comment by tvami** [comment]
Sorry, I'm on vacation this week. Please ping the other conveners, @cms-sw/alca-l2

**Comment by bsunanda** [comment]
@yuanchao @pohsun @francescobrivio Please approve this PR

**Comment by francescobrivio** [comment]
+alca

File: Calibration/IsolatedParticles/src/ChargeIsolation.cc [hunk 1]
```diff
@@ -361,4 +361,232 @@ namespace spr {
...
+                         edm::Handle<reco::TrackCollection> trkCollection,
+                         const CaloSubdetectorGeometry* gEB,
+                         const CaloSubdetectorGeometry* gEE,
+                         TrackDetectorAssociator& associator,
+                         TrackAssociatorParameters& parameters_,
+                         int ieta,
+                         int iphi,
+                         const std::string& theTrackQuality,
+                         bool
+#ifdef EDM_ML_DEBUG
```
Comment (perrotta) [inline Calibration/IsolatedParticles/src/ChargeIsolation.cc:377]: I miss the point for having both `EDM_ML_DEBUG` and `bool debug` in this file

Comment (bsunanda) [reply Calibration/IsolatedParticles/src/ChargeIsolation.cc:377]: I shall remove out in a subsequent PR. At that time I shall get rid of EDM_ML_DEBUG statements

Comment (bsunanda) [reply Calibration/IsolatedParticles/src/ChargeIsolation.cc:377]: I meant removing cost statements

Comment (perrotta) [reply Calibration/IsolatedParticles/src/ChargeIsolation.cc:377]: For such a simple update, why not profiting of this PR and making everything at once?

Comment (bsunanda) [reply Calibration/IsolatedParticles/src/ChargeIsolation.cc]: Taking care of this

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
+1
- (@francescobrivio changes since last alca signature are minimal and only related to commented out parts of the code, I assume you can confirm it)

**Comment by perrotta** [comment]
merge

**Comment by francescobrivio** [comment]
+alca

**Pull request merged**